ConnectorAutocadCivil/ConnectorAutocadCivil/UI/ConnectorBindingsAutocadCivil2.cs

```diff
@@ -0,0 +1,700 @@
+
+    public override string GetActiveViewName() => "Entire Document";
+
+    public override List<string> GetObjectsInView() // this returns all visible doc objects.
+    {
+      var objs = new List<string>();
+      using (Transaction tr = Doc.Database.TransactionManager.StartTransaction())
+      {
+        BlockTableRecord modelSpace = Doc.Database.GetModelSpace();
+        foreach (ObjectId id in modelSpace)
+        {
+          var dbObj = tr.GetObject(id, OpenMode.ForRead);
+          if (dbObj.Visible())
+            objs.Add(dbObj.Handle.ToString());
+        }
+        tr.Commit();
+      }
+      return objs;
+    }
+
+    public override List<string> GetSelectedObjects()
+    {
+      var objs = new List<string>();
+      if (Doc != null)
+      {
+        PromptSelectionResult selection = Doc.Editor.SelectImplied();
+        if (selection.Status == PromptStatus.OK)
+          objs = selection.Value.GetHandles();
+      }
+      return objs;
+    }
+
+    public override List<ISelectionFilter> GetSelectionFilters()
+    {
+      var layers = new List<string>();
+      if (Doc != null)
+      {
+        using (Transaction tr = Doc.Database.TransactionManager.StartTransaction())
+        {
+          LayerTable lyrTbl = tr.GetObject(Doc.Database.LayerTableId, OpenMode.ForRead) as LayerTable;
+          foreach (ObjectId objId in lyrTbl)
+          {
+            LayerTableRecord lyrTblRec = tr.GetObject(objId, OpenMode.ForRead) as LayerTableRecord;
+            layers.Add(lyrTblRec.Name);
+          }
+          tr.Commit();
+        }
+      }
+      return new List<ISelectionFilter>()
+      {
+        new ManualSelectionFilter(),
+        new ListSelectionFilter {Slug="layer",  Name = "Layers", Icon = "LayersTriple", Description = "Selects objects based on their layers.", Values = layers },
+        new AllSelectionFilter {Slug="all",  Name = "All", Icon = "CubeScan", Description = "Selects all document objects." }
+      };
+    }
+
+    //TODO
+    public override List<MenuItem> GetCustomStreamMenuItems()
+    {
+      return new List<MenuItem>();
+    }
+
+    public override void SelectClientObjects(string args)
+    {
+      throw new NotImplementedException();
+    }
+
+    #endregion
+
+    #region receiving 
+    public override async Task<StreamState> ReceiveStream(StreamState state, ProgressViewModel progress)
+    {
+      var kit = KitManager.GetDefaultKit();
+      var converter = kit.LoadConverter(Utils.AutocadAppName);
+      if (converter == null)
+        throw new Exception("Could not find any Kit!");
+      var transport = new ServerTransport(state.Client.Account, state.StreamId);
+
+      var stream = await state.Client.StreamGet(state.StreamId);
+
+      if (progress.CancellationTokenSource.Token.IsCancellationRequested)
+      {
+        return null;
+      }
+
+      if (Doc == null)
+      {
+        converter.Report.LogOperationError(new Exception($"No Document is open."));
```

Comment:
Here we should use the `progress.Report` and not `converter.Report` so that we can capture the errors even if the two reports have not been merged yet